Flaky: ZeebeClientTest.shouldUseCustomExecutorWithJobWorker

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? once thus far
- Does it block your work? no
- Do we suspect that it is a real failure? no

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
2023-07-17T11:57:27.4940642Z [ERROR] io.camunda.zeebe.client.ZeebeClientTest.shouldUseCustomExecutorWithJobWorker -- Time elapsed: 0.904 s <<< FAILURE!
2023-07-17T11:57:27.4941755Z java.lang.AssertionError: 
2023-07-17T11:57:27.4942416Z 
2023-07-17T11:57:27.4943034Z Expecting actual:
2023-07-17T11:57:27.4943438Z   0L
2023-07-17T11:57:27.4943822Z to be greater than:
2023-07-17T11:57:27.4944297Z   0L
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

<details><summary>Logs</summary>
<pre>
2023-07-17T11:57:27.4940642Z [ERROR] io.camunda.zeebe.client.ZeebeClientTest.shouldUseCustomExecutorWithJobWorker -- Time elapsed: 0.904 s <<< FAILURE!
2023-07-17T11:57:27.4941755Z java.lang.AssertionError: 
2023-07-17T11:57:27.4942416Z 
2023-07-17T11:57:27.4943034Z Expecting actual:
2023-07-17T11:57:27.4943438Z   0L
2023-07-17T11:57:27.4943822Z to be greater than:
2023-07-17T11:57:27.4944297Z   0L
2023-07-17T11:57:27.4944531Z 
2023-07-17T11:57:27.4945652Z 	at io.camunda.zeebe.client.ZeebeClientTest.shouldUseCustomExecutorWithJobWorker(ZeebeClientTest.java:426)
2023-07-17T11:57:27.4946702Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-07-17T11:57:27.4947942Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-07-17T11:57:27.4949183Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-07-17T11:57:27.4950192Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2023-07-17T11:57:27.4951078Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2023-07-17T11:57:27.4952147Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2023-07-17T11:57:27.4953275Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2023-07-17T11:57:27.4954403Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2023-07-17T11:57:27.4955500Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2023-07-17T11:57:27.4956414Z 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
2023-07-17T11:57:27.4957518Z 	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:258)
2023-07-17T11:57:27.4958483Z 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
2023-07-17T11:57:27.4959314Z 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
2023-07-17T11:57:27.4960140Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2023-07-17T11:57:27.4960952Z 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2023-07-17T11:57:27.4961839Z 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2023-07-17T11:57:27.4962726Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2023-07-17T11:57:27.4963774Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2023-07-17T11:57:27.4964653Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2023-07-17T11:57:27.4965406Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2023-07-17T11:57:27.4966193Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2023-07-17T11:57:27.4967118Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2023-07-17T11:57:27.4967908Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2023-07-17T11:57:27.4968651Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2023-07-17T11:57:27.4969397Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2023-07-17T11:57:27.4970126Z 	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
2023-07-17T11:57:27.4970783Z 	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
2023-07-17T11:57:27.4971636Z 	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
2023-07-17T11:57:27.4972778Z 	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
2023-07-17T11:57:27.4973794Z 	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
2023-07-17T11:57:27.4974998Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
2023-07-17T11:57:27.4976342Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
2023-07-17T11:57:27.4977884Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
2023-07-17T11:57:27.4979585Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
2023-07-17T11:57:27.4981038Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
2023-07-17T11:57:27.4982925Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
2023-07-17T11:57:27.4984234Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
2023-07-17T11:57:27.4985247Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
2023-07-17T11:57:27.4986420Z 	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
2023-07-17T11:57:27.4987693Z 	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
2023-07-17T11:57:27.4988938Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:191)
2023-07-17T11:57:27.4989982Z 	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
2023-07-17T11:57:27.4991068Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:186)
2023-07-17T11:57:27.4992525Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
2023-07-17T11:57:27.4993942Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:118)
2023-07-17T11:57:27.4995161Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
2023-07-17T11:57:27.4996200Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
2023-07-17T11:57:27.4997231Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
2023-07-17T11:57:27.4998155Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
2023-07-17T11:57:27.4998656Z 
</pre>
</details>

--- a/clients/java/src/test/java/io/camunda/zeebe/client/ZeebeClientTest.java
+++ b/clients/java/src/test/java/io/camunda/zeebe/client/ZeebeClientTest.java
@@ -25,6 +25,10 @@ import static io.camunda.zeebe.client.impl.ZeebeClientBuilderImpl.PLAINTEXT_CONN
 import static io.camunda.zeebe.client.impl.util.DataSizeUtil.ONE_MB;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
 
 import io.camunda.zeebe.client.api.worker.JobWorker;
 import io.camunda.zeebe.client.impl.NoopCredentialsProvider;
@@ -39,6 +43,7 @@ import java.util.Properties;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
@@ -410,7 +415,8 @@ public final class ZeebeClientTest extends ClientTest {
   @Test
   public void shouldUseCustomExecutorWithJobWorker() {
     // given
-    final ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(1);
+    final ScheduledThreadPoolExecutor executor = spy(new ScheduledThreadPoolExecutor(1));
+    final Duration pollInterval = Duration.ZERO;
     try (final ZeebeClient client =
             ZeebeClient.newClientBuilder().jobWorkerExecutor(executor).build();
         final JobWorker ignored =
@@ -418,12 +424,11 @@ public final class ZeebeClientTest extends ClientTest {
                 .newWorker()
                 .jobType("type")
                 .handler((c, j) -> {})
-                .pollInterval(Duration.ZERO)
+                .pollInterval(pollInterval)
                 .open()) {
-      // when - then - polling should be scheduled immediately on open, but may fail faster than
-      // we assert here, so we need to check both the scheduled count and the completed task count
-      // to see if anything has happened
-      assertThat(executor.getActiveCount() + executor.getCompletedTaskCount()).isPositive();
+      // when - then
+      verify(executor)
+          .schedule(any(Runnable.class), eq(pollInterval.toMillis()), eq(TimeUnit.MILLISECONDS));
     }
   }
 }

